src/facade/dragonfly_connection.cc

```diff
@@ -979,45 +993,21 @@ void Connection::DispatchFiber(util::FiberSocketBase* peer) {
     bool threshold_reached = dispatch_q_cmds_count_ > squashing_threshold;
     bool are_all_plain_cmds = dispatch_q_cmds_count_ == dispatch_q_.size();
     if (squashing_enabled && threshold_reached && are_all_plain_cmds) {
-      vector<CmdArgList> args;
-      args.reserve(dispatch_q_.size());
-      for (auto& msg : dispatch_q_) {
-        CHECK(holds_alternative<PipelineMessagePtr>(msg.handle));
-        auto& pipe_msg = get<PipelineMessagePtr>(msg.handle);
-        args.push_back(absl::MakeSpan(pipe_msg->args));
-      }
-
-      cc_->async_dispatch = true;
-      service_->DispatchManyCommands(absl::MakeSpan(args), cc_.get());
-      cc_->async_dispatch = false;
-
-      // Flush strictly before the dispatch queue is cleared so that no sync dispatch can occur
-      if (dispatch_q_.size() == args.size()) {  // Flush if no new messages appeared
-        builder->FlushBatch();
-        builder->SetBatchMode(false);  // in case the next dispatch is sync
-      }
-
-      DCHECK(!cc_->sync_dispatch);
-      // Dispatch queue could have grown, so handle strictly as many as we executed
-      for (size_t i = 0; i < args.size(); i++) {
-        recycle(move(dispatch_q_.front()));
-        dispatch_q_.pop_front();
-      }
```

Comment:
The bug here is that we don't account for commands that have been inserted upfront